src/client.c

```diff
@@ -894,9 +896,39 @@ static bool handle_client_work(PgSocket *client, PktHdr *pkt)
 	/* forward the packet */
 	sbuf_prepare_send(sbuf, &client->link->sbuf, pkt->len);
 
-	/* log the query, if needed */
-	if (cf_log_packets)
-		log_pkt_to_buffer(pkt, client);
+	if (cf_log_packets) {
+		if (pkt->type == 'Q' || pkt->type == 'P' || pkt->type == 'B' || pkt->type == 'E') {
+			log_pkt_to_buffer(pkt, client);
+			
+			// Check packet incomplete and assign values to struct
+			if (cf_buffer_incomplete_packets) {
+				if (incomplete_pkt(pkt)) {
+					if ((int)pkt->len > (int)cf_sbuf_len) {
+						client->incomplete_packet_buffer.started = true;
+						client->incomplete_packet_buffer.pkt_data = &pkt->data;
```

Comment:
so this won't do what we want it to do, i think, unless pkt->data isn't cleared after a send, which I _think_ it does (please correct me if wrong).

so what you want here is actually a `memcpy()` of `pkt_data` to `.(.._buffer.pkt_data + ..._buffer.pkt_len)` (basically to memcpy + append the data to the end of the buffer, unless there is a helper for mbuf for that), and I assume you would want to `+= pkt->len` as well to the total size.